front/components/workspace/ProviderManagementModal.tsx

```diff
@@ -146,10 +161,15 @@ export function ProviderManagementModal({
 
   const hasChanges =
     !isEqual(providerStates, initialProviderStates) ||
-    embeddingProvider !== owner.defaultEmbeddingProvider;
+    embeddingProvider !== workspace?.defaultEmbeddingProvider;
```

Comment:
I'm not sure if it's only me but it took me a bit of time to understand how things work in this component (basically to understand `providerStates` & `embeddingProvider` = local state, `initialProviderStates` & `workspace?.defaultEmbeddingProvider` = computed from or direct server data).

Was it easy for you to understand? I wonder if we can add some comments maybe?

Reply:
Agreed, added comments